src/cca/scripts/cloud_socks_proxy_server.ts

```diff
@@ -0,0 +1,88 @@
+import uparams = require('uparams');
+import * as jsurl from 'jsurl';
+
+import * as net from '../../lib/net/net.types';
+import * as social from '../../interfaces/social';
+import * as uproxy_core_api from '../../interfaces/uproxy_core_api';
+
+import CoreConnector from '../../generic_ui/scripts/core_connector';
+import { SocksProxyServer } from '../model/socks_proxy_server';
+
+
+// A local Socks server that provides access to a remote uProxy Cloud server via RTC.
+export class CloudSocksProxyServer implements SocksProxyServer {
+  private instancePath_: social.InstancePath;
+
+  // Constructs a server that will use the given CoreApi to start the local proxy.
+  // It takes the IP address of the uProxy cloud server it will use for Internet access.    
+  public constructor(private core_: uproxy_core_api.CoreApi,
+                     private remoteIpAddress_: string) {
+    this.instancePath_ = {
+      network: {
+        name: 'Cloud',
+        userId: 'me'
+      },
+      userId: remoteIpAddress_,
+      instanceId: remoteIpAddress_
+    }
+  }
+
+  public remoteIpAddress(): string {
```

Comment:
Could you clarify?
That's a basic Typescript construct. I don't see why we would ban it.